server/base/src/main/java/org/apache/accumulo/server/conf/util/ZooPropSetTool.java

```diff
@@ -0,0 +1,391 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.server.conf.util;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.accumulo.server.conf.util.ZooPropUtils.getInstanceId;
+import static org.apache.accumulo.server.conf.util.ZooPropUtils.getNamespaceIdToNameMap;
+import static org.apache.accumulo.server.conf.util.ZooPropUtils.getTableIdToName;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import org.apache.accumulo.core.cli.ConfigOpts;
+import org.apache.accumulo.core.conf.Property;
+import org.apache.accumulo.core.data.InstanceId;
+import org.apache.accumulo.core.data.NamespaceId;
+import org.apache.accumulo.core.data.TableId;
+import org.apache.accumulo.core.fate.zookeeper.ZooReader;
+import org.apache.accumulo.core.fate.zookeeper.ZooReaderWriter;
+import org.apache.accumulo.server.ServerContext;
+import org.apache.accumulo.server.conf.codec.VersionedProperties;
+import org.apache.accumulo.server.conf.store.NamespacePropKey;
+import org.apache.accumulo.server.conf.store.PropStoreKey;
+import org.apache.accumulo.server.conf.store.SystemPropKey;
+import org.apache.accumulo.server.conf.store.TablePropKey;
+import org.apache.accumulo.server.conf.store.impl.PropStoreWatcher;
+import org.apache.accumulo.server.conf.store.impl.ReadyMonitor;
+import org.apache.accumulo.server.conf.store.impl.ZooPropStore;
+import org.apache.accumulo.server.util.PropUtil;
+import org.apache.accumulo.start.spi.KeywordExecutable;
+import org.apache.zookeeper.KeeperException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.beust.jcommander.Parameter;
+import com.google.auto.service.AutoService;
+
+@AutoService(KeywordExecutable.class)
+public class ZooPropSetTool implements KeywordExecutable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ZooPropSetTool.class);
+  private final NullWatcher nullWatcher =
+      new NullWatcher(new ReadyMonitor(ZooInfoViewer.class.getSimpleName(), 20_000L));
+
+  /**
+   * No-op constructor - provided so ServiceLoader autoload does not consume resources.
+   */
+  public ZooPropSetTool() {}
+
+  public static void main(String[] args) throws Exception {
+    new ZooPropSetTool().execute(args);
+  }
+
+  @Override
+  public String keyword() {
+    return "zoo-prop-set-tool";
```

Comment:
Will revisit and review this later, but wanted to make this one initial suggestion:

```suggestion
    return "zoo-prop-editor";
```

Reply:
I don't think the term "editor" implies interactivity... just that it's a tool to make edits. There are plenty of editors that aren't interactive (`sed` comes immediately to mind, among a few others). Its function is to edit. Its help/usage will quickly resolve any confusion over *how* it performs the editing function.

Mainly, I agree with you about `zoo-prop-set-tool` being unwieldy, but also, the current name is misleading, because it can do more kinds of edits than merely setting a property. It can modify a property, and delete one, too. The term "editor" implies any kind of edit, and seems like the most appropriate here.

Reply:
I push up a change with it called editor soon.  Your suggestion is a trap ;-)  Need to update KeywordExecatableIT too.

Reply:
Addressed in e2d827213b